### PR TITLE
undef(resolve) adding Hoa namespace to resolve to avoid conflict

### PR DESCRIPTION
**Pull request by Grummfy**
related to https://github.com/hoaproject/Protocol/issues/18
once we have discuted about it, I will get back on other repository depending on this function.

note with php 5.6+ we can use ```use function \Hoa\resolve``` in the import part

### CODE REVIEW

**Comment by vonglasow** [comment]
Can you complete also the README please ?

File: Wrapper.php [hunk 1]
```diff
@@ -583,7 +583,7 @@ public function getStreamName()
 
 }
 
-namespace
```
Comment (Hywan) [inline Wrapper.php]: Remove this.

File: Test/Unit/Wrapper.php [hunk 2]
```diff
@@ -902,7 +902,7 @@ protected function openDirectory(SUT $wrapper, array $children = [])
         $wrapper->dir_opendir('hoa://Test/Vfs/Bar?type=directory', 0);
 
         foreach ($children as $child) {
-            resolve('hoa://Test/Vfs/Bar/' . $child . '?type=file');
```
Comment (Hywan) [inline Test/Unit/Wrapper.php]: `Hoa\Protocol\Protocol::getInstance()->resolve(…)` is better.

File: Bin/Resolve.php [hunk 3]
```diff
@@ -148,7 +148,7 @@ public function main()
                 ' is equivalent to:', "\n";
         }
 
-        $resolved = resolve($path, $exists, $unfold);
```
Comment (Hywan) [inline Bin/Resolve.php]: `Hoa\Protocol\Protocol::getInstance()->resolve(…)` is better.

**Review by Hywan — CHANGES_REQUESTED** [review]
I guess it's cool with this modification. Thoughts?

**Comment by Grummfy** [comment]
here it is

**Comment by Grummfy** [comment]
Yes, because less bc break is good.
No, because it break my laravel app ;)

But, I can wait. What do we need for breaking php compatibility?
Did I update other library to remove it?

**Comment by Hywan** [comment]
We must implement `Hoa\Option`, and then use it for all new libraries. With `Hoa\Option`, we will introduce BC, and thus we will upgrade to PHP 7. I can focus my work on `Hoa\Option`, then use it in `Hoa\Consistency`, do a new major release, and use it in `Hoa\Protocol` with a new major release. Thoughts?

**Comment by Grummfy** [comment]
OK.

**Comment by drwmrrs** [comment]
Hey all! 

We've just implemented the Hoa\Ruler package in Laravel and this is causing some serious headaches for us as we have other packages using Laravel's `resolve` function internally.

Is there any update on when this will be released? Is it safe to just use `dev-master` for the protocol package for now?

**Comment by jaroslavas** [comment]
Hey guys, when will you release this? Your libraries are still unusable with Laravel because of this resolve function!

**Pull request merged**